### PR TITLE
Remove obsolete processes

### PR DESCRIPTION
**Pull request by jkokosar**
## Checklist

* [ ] Update CHANGELOG.rst for each commit separately:
  * Pay attention to write entries under the "Unreleased" section.
  * Mark all breaking changes as "**BACKWARD INCOMPATIBLE:**" and put them
    before non-breaking changes.
  * If a commit modifies a feature listed under "Unreleased" section,
    it might be sufficient to modify the existing CHANGELOG entry from previous
    commit(s).
* [ ] Bump the process version:
  * **MAJOR version (first number)**: Backward incompatible changes (changes
    that break the api/interface). Examples: renaming the input/output, adding
    mandatory input, removing input/output...
  * **MINOR version (middle number)**: add functionality or changes in a
    backwards-compatible manner. Examples: add output field, add non-mandatory
    input parameter, use a different tool that produces same results...
  * **PATCH version (last number)**: changes/bug fixes that do not affect
    the api/interface. Examples: typo fix, change/add warning messages...
* [ ] All inputs are used in process.
* [ ] All output fields have their ``re-save`` calls.

### Processes ###

* Code that was once accepted should be kept like it is. It should be changed
  only when its functionality has changed. Small parts may be changed
  to fix bugs (by changing code only to make it nicer you might introduce a bug).
* Set the optimal number of cores that the tool can use (check if parallelization
  can be performed).
* Description of tool should be self explanatory with links to tool's homepage
  and publication.
* Write explanatory input labels and add short command line parameter names
  in square brackets (for example: Maximum length [``--maxlength``]).
* Where possible provide default values for the process input fields.
  If they are provided, then ``required: false`` is not needed.
* Add descriptive error and warning messages.
* Place a ``re-checkrc`` call after each process step (call of a tool, a
  Bash command, a Python script or an R script).
* Add ``re-progress`` calls where appropriate.

### Commit and PR messages ###

* Each commit should be minimal (i.e. 1 change) and self-contained (including
  tests).
* Changes to Docker images need to be accepted in a separate pull request.
  Before the new images can be used for automated testing of your PR, they
  must be manually tagged and pushed to Docker Hub.
* Mark incomplete PRs with the [WIP] (Work In Progress) tag.
* Do not paste links to private repositories from the public one.

### Tests ###

* Add meaningful names of test files with spaces:
  * Genomes: ``genome species.fa.gz``
  * Annotations: ``annotation species.gtf.gz``
  * Adapters: ``adapters-name/type-source``.
* Output and specific input files should include the name of the tested tool.
* Keep test files' size small. If there is no other option, add the large
  test file to the ``files/large`` directory so it will be tracked with Git LFS.
* Pay attention to edge cases: what will happen if any data is
  missing, if length of some list is 0, if string is given where one expects
  integer, if file is empty, ...
* When testing workflows, try to verify workflow success by checking one
  of the last steps. Pick the one that is most likely to fail. Also test
  that all steps have 'OK' status.

### CODE REVIEW

File: resolwe_bio/processes/workflows/amplicon.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@
   data_name: "{{ reads|sample_name|default('?') }}"
   requirements:
     expression-engine: jinja
-  version: 3.0.1
+  version: 3.1.0
```
Comment (JureZmrzlikar) [inline resolwe_bio/processes/workflows/amplicon.yml]: IMO, removing a process from workflow is equal to removing an output from the process, which is a breaking change. Raise version accordingly and prepend changelog entry with "backward-incompatible"...

File: resolwe_bio/tests/processes/test_expression.py [hunk 2]
```diff
@@ -18,13 +18,13 @@ def test_cufflinks(self):
...
 
-            inputs = {
+            aligned_reads = self.run_process('alignment-hisat2', {
                 'genome': genome.pk,
                 'reads': reads.pk,
-                'annotation': annotation_gtf.pk,
-                'PE_options': {
-                    'library_type': "fr-unstranded"}}
-            aligned_reads = self.run_process('alignment-tophat2', inputs)
+                'spliced_alignments': {
```
Comment (JureZmrzlikar) [inline resolwe_bio/tests/processes/test_expression.py]: How is this change connected to the process removal?

File: resolwe_bio/tests/workflows/test_heat_seq.py [hunk 3]
```diff
@@ -1,53 +0,0 @@
...
-    @tag_process('workflow-heat-seq')
-    def test_heatseq_workflow(self):
-        with self.preparation_stage():
-            inputs = {
-                'src1': ['heat_seq_mate1.fq.gz'],
-                'src2': ['heat_seq_mate2.fq.gz']}
-            reads = self.run_process('upload-fastq-paired', inputs)
-
-            inputs = {
-                'src': 'chr21_small.fasta.gz',
```
Comment (JureZmrzlikar) [inline resolwe_bio/tests/workflows/test_heat_seq.py]: Also remove this file.

File: resolwe_bio/tests/processes/test_expression.py (hunk 2 above)
Comment (jkokosar) [reply resolwe_bio/tests/processes/test_expression.py]: `alignment-tophat2` was removed as a process and was used in a few of the tests. `alignment-hisat2` is more modern replacement aligner to be used in this workflows, so this replacement was made.

File: docs/CHANGELOG.rst [hunk 4]
```diff
@@ -12,6 +12,18 @@ Unreleased
 
 Changed
 -------
+- **BACKWARD INCOMPATIBLE:** Remove obsolete processes: ``bsmap``,
```
Comment (JenkoB) [inline docs/CHANGELOG.rst]: Great cleaning!! Did you remove ``wgbs.yml`` workflow? It should be also removed and stated in changelog.

**Pull request merged**